towerlib/entities/job.py

```diff
@@ -649,8 +657,68 @@ def job_events(self):
                              primary_match_field='name',
                              url=url)
 
+    def monitor(self, timeout=None, interval=0.25):
+        """Monitor the job execution events.
+
+        Returns:
+            string: The job status of the job
+
+        """
+        next_line = 0
+        started = time.time()
+        while self.status not in ["running"]:
+            time.sleep(interval)
+            logging.info(f"... {self.status}")
+
+        logging.info("Starting Job Output Stream")
+        while True:
+            if timeout and time.time() - started > timeout:
+                logging.error("Monitoring aborted due to timeout.")
+                break
+
+            next_line = self._fetch(next_line)
+
+            time.sleep(interval)
+            if self.event_processing_finished is True or self.status in ("error", "cancelled"):
+                self._fetch(next_line)
+                break
+
+        return self.status
+
+    def _fetch(self, next_line):
```

Comment:
This fetch method is the same as the fetch method in line 1786 with the difference that in that one the job events are passed as an argument. If this accepts the arguments also they are identical. 

Maybe this is something that would need to be modelled in an JobEntity that provides that capability and then these objects that need that can can inherit from that. 

Let's ponder a bit on this...

Reply:
Yeah, i figured the fetch model would make more sense when modelled as a JobEntity to support inheritance.. unsure on the best approach here.